zerver/migrations/0498_set_default_value_for_realm_direct_message_initiator_group_and_more.py

```diff
@@ -0,0 +1,75 @@
+# Generated by Django 4.2.8 on 2024-01-01 10:03
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def set_default_value_for_direct_message_initiator_group(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    UserGroup = apps.get_model("zerver", "UserGroup")
+
+    EVERYONE_GROUP_NAME = "role:everyone"
+    NOBODY_GROUP_NAME = "role:nobody"
+
+    for realm in Realm.objects.all():
+        if realm.direct_message_initiator_group is not None:
+            continue
+        if realm.private_message_policy == 2:
+            nobody_group = UserGroup.objects.get(
+                name=NOBODY_GROUP_NAME, realm=realm, is_system_group=True
+            )
+            realm.direct_message_initiator_group = nobody_group
+            realm.save(update_fields=["direct_message_initiator_group"])
+        else:
+            everyone_group = UserGroup.objects.get(
+                name=EVERYONE_GROUP_NAME, realm=realm, is_system_group=True
+            )
+            realm.direct_message_initiator_group = everyone_group
+            realm.save(update_fields=["direct_message_initiator_group"])
+
+
+def set_default_value_for_direct_message_permission_group(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    UserGroup = apps.get_model("zerver", "UserGroup")
+
+    EVERYONE_GROUP_NAME = "role:everyone"
+    NOBODY_GROUP_NAME = "role:nobody"
+    for realm in Realm.objects.all():
+        if realm.direct_message_permission_group is not None:
+            continue
+        if realm.private_message_policy == 2:
+            nobody_group = UserGroup.objects.get(
+                name=NOBODY_GROUP_NAME, realm=realm, is_system_group=True
+            )
+            realm.direct_message_permission_group = nobody_group
+            realm.save(update_fields=["direct_message_permission_group"])
+        else:
+            everyone_group = UserGroup.objects.get(
+                name=EVERYONE_GROUP_NAME, realm=realm, is_system_group=True
+            )
+            realm.direct_message_permission_group = everyone_group
+            realm.save(update_fields=["direct_message_permission_group"])
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("zerver", "0497_realm_direct_message_initiator_group_and_more"),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            set_default_value_for_direct_message_initiator_group,
+            elidable=True,
+            reverse_code=migrations.RunPython.noop,
+        ),
+        migrations.RunPython(
+            set_default_value_for_direct_message_permission_group,
+            elidable=True,
+            reverse_code=migrations.RunPython.noop,
+        ),
```

Comment:
I would merge the two functions, and just have a single migration step that sets the defaults for both groups.

Reply:
Done in the recent changes.